aikit/automl/controller.py

```diff
@@ -0,0 +1,287 @@
+        else:
+            X, y, *_ = self.data.load_pickle(self.data_key)
+            self.automl_config = AutoMlConfig()
+            self.automl_config.guess_everything(X, y)
+            self.job_config = JobConfig(self.automl_config.type_of_problem)
+            self.job_config.guess_cv(n_splits=5)
+            self.job_config.guess_scoring()
+
+            self.storage.save_pickle(self.automl_config, 'automl_config')
+            self.storage.save_pickle(self.job_config, 'job_config')
+            self.storage.save_json(self.status, 'status')
+
+    def _init_generators(self):
+        self.automl_guider = AutoMlModelGuider(self.job_config)
+        self.generator = RandomModelGenerator(self.automl_config, random_state=self.random_state)
+
+        if self.job_config.start_with_default:
+            self._default_iterator = self.generator.iterator_default_models()
+            self._default_iterator_empty = False
+
+        if self.job_config.do_blocks_search:
+            self._block_search_iterator = self.generator.iterate_block_search(random_order=True)
+            self._block_search_iterator_empty = False
+
+    def run(self):
+        while self.queue.size() < self.queue_size:
+            self.create_job()
+
+        logger.info('Queue is full. Adding new jobs every {} seconds'.format(self.jobs_update_time))
+        while True:
+            current_queue_size = self.queue.size()
+            for _ in range(self.queue_size - current_queue_size):
+                self.create_job()
+            time.sleep(self.jobs_update_time)
+
+    def create_job(self):
+        job = self.get_next_job()
+
+        if job['job_id'] in self.reader.jobs:
+            return
+
+        job = self.compute_job_cv_type(job)
+        job = self.compute_job_metric_threshold(job)
+
+        self.storage.save_json(self.status, 'status')
+        self.storage.save_special_json(job, job['job_id'], 'jobs')
+        self.queue.enqueue(job['job_id'])
+        logger.info('Generating {} job [{}]'.format(job['job_type'], job['job_id']))
+
+    def get_next_job(self):
+        # iterate on default iterator
+        if self.status['mode'] == 'default' and not self._default_iterator_empty:
+            try:
+                next_job = next(self._default_iterator)
+                return self.get_job('default', *next_job)
+            except StopIteration:
+                logger.info("Default iterator cleared out")
+                self._default_iterator_empty = True
+                self.status['mode'] = 'random'
+
+        # iterate on block search iterator
+        if not self._block_search_iterator_empty and self.random_state.rand(1)[0] <= self.p_block_search:
+            try:
+                next_job = next(self._block_search_iterator)
+                return self.get_job('block-search', *next_job)
+            except StopIteration:
+                logger.info("Block-search iterator cleared out")
+                self._block_search_iterator_empty = True
```

Comment:
I think we should use same logic for 'default' and 'block' and store both information within 'status'
maybe just storing the 2 boolean instead of the 1 status ?

Reply:
Makes the process more clear